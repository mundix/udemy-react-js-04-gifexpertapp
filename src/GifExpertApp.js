import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon ball'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon ball']);

    // const handleAdd = (name = 'HunterXHuner') => {
    //     setCategories( cats => [...cats, name]);
    // }

    // La funcion setCategories, se pasa como argumento , y es una referencia por eso 
    // Me permite acceder a categories para asignarlo 
    return (
        <>
          <h2>Gif Expert App</h2>  
          <AddCategory setCategories={setCategories}/>
          <hr />
        
          <ol>
              { categories.map( (category, index) => {
                  return <li key={category}>{category}</li>;
              })}
          </ol>
        </>
    )
}
