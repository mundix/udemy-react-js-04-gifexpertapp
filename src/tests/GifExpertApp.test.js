import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas en archivo GifExpertApp', () => {
    
   test('Comprobar GifExpertApp correctamente', () => {
       
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
   })
    
});
