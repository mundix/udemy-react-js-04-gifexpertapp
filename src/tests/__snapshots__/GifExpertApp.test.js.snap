// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en archivo GifExpertApp Comprobar GifExpertApp correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch Men"
      key="One Punch Men"
    />
  </ol>
</Fragment>
`;
